src/stringify.js

```diff
@@ -2,9 +2,15 @@
 
 /* global toDebugString: true */
 
-function serializeObject(obj) {
+function serializeObject(obj, maxDepth) {
   var seen = [];
 
+  // There is no direct way to stringify object until reaching a specific depth
+  // and a very deep object can cause a performance issue, so we copy the object
+  // based on this specific depth and then stringify it.
+  if (maxDepth > 0) {
```

Comment:
Unlikely to happen, but in order to avoid unnecessry copying (e.g. if `maxDepth === '1'), let's be consistent with this condition is all places (e.g. `copy()`, `serializeObject()`, `configureErrorHandling()`).